sqlite_data_norm.c

```diff
@@ -0,0 +1,303 @@
+/*-------------------------------------------------------------------------
+ *
+ * SQLite Foreign Data Wrapper for PostgreSQL
+ *
+ * SQLite functions for data normalization
+ * This function is useful for mixed affinity inputs for PostgreSQL
+ * data column. Also some UUID functions are implemented here according 
+ * the uuid SQLite exension, Public Domain
+ * https://www.sqlite.org/src/file/ext/misc/uuid.c
+ *
+ * IDENTIFICATION
+ * 		sqlite_data_norm.c
+ *
+ *-------------------------------------------------------------------------
+ */
+
+/*
+ * This SQLite extension implements functions that handling RFC-4122 UUIDs
+ * Three SQL functions are implemented:
+ *
+ *	 gen_random_uuid() - generate a version 4 UUID as a string
+ *	 uuid_str(X)	   - convert a UUID X into a well-formed UUID string
+ *	 uuid_blob(X)	  - convert a UUID X into a 16-byte blob
+ *
+ * The output from gen_random_uuid() and uuid_str(X) are always well-formed
+ * RFC-4122 UUID strings in this format:
+ *
+ *		xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx
+ *
+ * All of the 'x', 'M', and 'N' values are lower-case hexadecimal digits.
+ * The M digit indicates the "version".  For uuid4()-generated UUIDs, the
+ * version is always "4" (a random UUID).  The upper three bits of N digit
+ * are the "variant".  This library only supports variant 1 (indicated
+ * by values of N between '8' and 'b') as those are overwhelming the most
+ * common.  Other variants are for legacy compatibility only.
+ *
+ * The output of uuid_blob(X) is always a 16-byte blob. The UUID input
+ * string is converted in network byte order (big-endian) in accordance
+ * with RFC-4122 specifications for variant-1 UUIDs.  Note that network
+ * byte order is *always* used, even if the input self-identifies as a
+ * variant-2 UUID.
+ *
+ * The input X to the uuid_str() and uuid_blob() functions can be either
+ * a string or a BLOB. If it is a BLOB it must be exactly 16 bytes in
+ * length or else a NULL is returned.  If the input is a string it must
+ * consist of 32 hexadecimal digits, upper or lower case, optionally
+ * surrounded by {...} and with optional "-" characters interposed in the
+ * middle.  The flexibility of input is inspired by the PostgreSQL
+ * implementation of UUID functions that accept in all of the following
+ * formats:
+ *
+ *	 A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11
+ *	 {a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11}
+ *	 a0eebc999c0b4ef8bb6d6bb9bd380a11
+ *	 a0ee-bc99-9c0b-4ef8-bb6d-6bb9-bd38-0a11
+ *	 {a0eebc99-9c0b4ef8-bb6d6bb9-bd380a11}
+ *
+ * If any of the above inputs are passed into uuid_str(), the output will
+ * always be in the canonical RFC-4122 format:
+ *
+ *	 a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11
+ *
+ * If the X input string has too few or too many digits or contains
+ * stray characters other than {, }, or -, then NULL is returned.
+ */
+#include <assert.h>
+#include <ctype.h>
+#include <string.h>
+
+#include "sqlite3.h"
+
+int sqlite_fdw_data_norm_functs_init(sqlite3* db);
```

Comment:
This line is unnecessary.

Reply:
Are you about line72 , @t-kataym  ?
In this case with PostgreSQL 16 + gcc 0.2.1 20210110
```
sqlite_data_norm.c:287:1: warning: no previous prototype for ‘sqlite_fdw_data_norm_functs_init’ [-Wmissing-prototypes]
  287 | sqlite_fdw_data_norm_functs_init(sqlite3* db) {
      |
```
`int sqlite_fdw_data_norm_functs_init(sqlite3* db);` in not in `sqlite_fdw.h` because there is only 1 function with outer usage in the file.

Reply:
Refactoring is possible, changed by adding `#include "sqlite_fdw.h"`.

Reply:
Fixed in https://github.com/pgspider/sqlite_fdw/pull/88/commits/72e1b22177db37251b2cda4474eb46955d1d6f26, please check again.

Reply:
> Are you about line72 , @t-kataym ?
> Refactoring is possible, changed by adding #include "sqlite_fdw.h".

Yes. Your refactoring is as my expectation. Thank you for fixing.